common-lib/src/main/scala/com/gu/mediaservice/lib/aws/S3.scala

```diff
@@ -168,14 +170,16 @@ class S3(config: CommonConfig) extends GridLogging {
 
 object S3Ops {
   // TODO make this localstack friendly
+  // TODO: Make this region aware - i.e. RegionUtils.getRegion(region).getServiceEndpoint(AmazonS3.ENDPOINT_PREFIX)
   private val s3Endpoint = "s3.amazonaws.com"
 
-  def buildS3Client(config: CommonConfig, localstackAware: Boolean = true): AmazonS3 = {
-    // Force v2 signatures: https://github.com/aws/aws-sdk-java/issues/372
-    // imgops proxies direct to S3, passing the AWS security signature as query parameters
-    // This does not work with AWS v4 signatures, presumably because the signature includes the host
+  def buildS3Client(config: CommonConfig, forceV2Sigs: Boolean = false, localstackAware: Boolean = true): AmazonS3 = {
+
     val clientConfig = new ClientConfiguration()
-    clientConfig.setSignerOverride("S3SignerType")
+    // Option to disable v4 signatures (https://github.com/aws/aws-sdk-java/issues/372) which is required by imgops
+    // which proxies direct to S3, passing the AWS security signature as query parameters. This does not work with
+    // AWS v4 signatures, presumably because the signature includes the host
+    if (forceV2Sigs) clientConfig.setSignerOverride("S3SignerType")
```

Comment:
minor: is it worth logging when we enter this path, just to get a sense of scale?

Reply:
Essentially it'll happen once per instance for a service that calls `signUrl` (but only once as it's a lazy val so client will be kept around after that). As far as I can see it's only the MediaApi that does this. I'm not sure it will tell us much that can't be inferred from a few minutes reading the source code.